### DISCUSSION TITLE
Toggling drop down with alpine js based on livewire computed property

### DISCUSSION THREAD

**Discussion by rico** [livewire/livewire#1368] [1. Help] [answered]
I have the following component view:

```
<div x-data="{ open: {{ $this->open }} }">
    <form class="form-inline mr-4 d-none d-sm-flex" @click.away="open = false">

        <div class="input-group input-group-flush input-group-merge" data-toggle="lists">

            <input type="search" class="form-control form-control-prepended dropdown-toggle"
                   wire:model.debounce.300ms="query"
                   x-on:focus="open = {{ $this->open }}"
                   placeholder="Suchen"
                   aria-label="Benutzer suchen" />

            <div class="input-group-prepend">
                <div class="input-group-text">
                    <i class="fad fa-users"></i>
                </div>
            </div>

            <div class="dropdown-menu dropdown-menu-card" :class="{ 'show': open }">

                <div class="card-body">

                    <div class="list-group list-group-flush list my-n3">

                        @foreach($users as $user)
                            <a class="list-group-item"  href="#" wire:click.prevent="handler">
                                <div class="row align-items-center">
                                    <div class="col ml-n2 text-body mb-1">
                                        {{ $user->name }}
                                    </div>
                                </div>
                            </a>
                        @endforeach
                    </div>
                </div>
            </div>
            
        </div>
    </form>
</div>
```

Where `$this->open` is a computed property. The expected behaviour is that when the user clicks away from the component, the dropdown should disappear - that works. Now, when the user clicks the input and `$this->open` is `true`, the dropdown should reappear - this does not work. 

If I set the the value of `open` to `true` in the `x-on:focus` handler, like `x-on:focus="open = true"` , the dropdown is shown on focus.

Any idea how I can get this to work?

The computed property method in the livewire component looks like this:

```
public function getOpenProperty()
{
    return count($this->users) > 0 ? 'true' : 'false';
}
```

Thanks!

**Comment by joshhanley** [comment, marked as answer]
@rico Alpine, once initialised, won't pick up new changes to the data rendered in blade. Hence why you're open computed property changes aren't being reflected.

You'd be better off creating a public property called `$open` and using `@entangle` to keep Livewire and Alpine in sync.

Like this 
```blade
<div x-data="{ open: @entangle('open')">
```

Hope this helps!

**Reply by rico** [reply]
Thanks @joshhanley for answering this question after such a long time. It has been a while since I worked with Alpine and Livewire, but I'm sure your solution works.

**Reply by joshhanley** [reply]
@rico oh! I didn't realise the issue was so old! For some reason I had a notification about it last week, so just answered it 😁
